src/Core/Domain/CmsPageCategory/Command/ToggleCmsPageCategoryStatusCommand.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\CmsPageCategory\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\CmsPageCategory\ValueObject\CmsPageCategoryId;
+
+/**
+ * Class ToggleCmsPageCategoryStatusCommand is responsible for turning on and off cms page category status.
+ */
+class ToggleCmsPageCategoryStatusCommand
+{
+    /**
+     * @var CmsPageCategoryId
+     */
+    private $cmsPageCategoryId;
+
+    /**
+     * @param CmsPageCategoryId $cmsPageCategoryId
+     */
+    public function __construct(CmsPageCategoryId $cmsPageCategoryId)
```

Comment:
After discussing it with eternoendless, we would like all Commands to only accept primitive php types as input. The idea is to make them as simple as possible to use.

One way to do it is to accept integers as input, but convert them into the right type so you benefit from ValueObject properties 😉 
```
public function __construct($cmsPageCategoryId)
    {
    	if (false === ($cmsPageCategoryId instanceof CmsPageCategoryId::class)) {
    		$cmsPageCategoryId = new cmsPageCategoryId($cmsPageCategoryId);
    	}

        $this->cmsPageCategoryId = $cmsPageCategoryId;
    }
```